lmp/model/_base_res_rnn_model.py

```diff
@@ -0,0 +1,235 @@
+            raise TypeError('`vocab_size` must be instance of `int`.')
+
+        # Value Check.
+        if d_emb < 1:
+            raise ValueError('`d_emb` must be bigger than or equal to `1`.')
+
+        if d_hid < 1:
+            raise ValueError('`d_hid` must be bigger than or equal to `1`.')
+
+        if not (0 <= dropout <= 1):
+            raise ValueError('`dropout` must range from `0.0` to `1.0`.')
+
+        if num_linear_layers < 1:
+            raise ValueError(
+                '`num_linear_layers` must be bigger than or equal to `1`.'
+            )
+
+        if num_rnn_layers < 1:
+            raise ValueError(
+                '`num_rnn_layers` must be bigger than or equal to `1`.'
+            )
+
+        # Embedding layer
+        # Dimension: (V, E)
+        self.emb_layer = torch.nn.Embedding(
+            num_embeddings=vocab_size,
+            embedding_dim=d_emb,
+            padding_idx=pad_token_id
+        )
+        self.emb_dropout = torch.nn.Dropout(dropout)
+
+
+        # Project d_emb to d_jid
+        # Dimension: (E, H)
+        self.proj_emb_to_hid = torch.nn.Sequential(
+            torch.nn.Linear(
+                in_features=d_emb,
+                out_features=d_hid
+            ),
+            torch.nn.Dropout(dropout)
+        )
+
+        # Sequential RNN layers.
+        # Dimension: (H, H)
+        rnn_blocks = []
+        for _ in range(num_rnn_layers):
+            rnn_blocks.append(
+                lmp.model.BaseResRNNBlock(
+                    d_hid=d_hid,
+                    dropout=dropout
+                )
+            )
+        rnn_blocks.append(
+            torch.nn.Dropout(dropout)
+        )
+        self.rnn_layer = torch.nn.Sequential(*rnn_blocks)
+
+        # Sequential linear layer.
+        # Dimension: (H, E)
+        proj_hid_to_emb = []
+        for _ in range(num_linear_layers):
+            proj_hid_to_emb.append(
+                torch.nn.Linear(
+                    in_features=d_hid,
+                    out_features=d_hid
+                )
+            )
+            proj_hid_to_emb.append(
+                torch.nn.ReLU()
+            )
+            proj_hid_to_emb.append(
+                torch.nn.Dropout(dropout)
+            )
+
+        proj_hid_to_emb.append(
+            torch.nn.Linear(
+                in_features=d_hid,
+                out_features=d_emb
+            )
+        )
+        self.proj_hid_to_emb = torch.nn.Sequential(*proj_hid_to_emb)
+
+    def forward(
+            self,
+            batch_sequences: torch.Tensor
+    ) -> torch.Tensor:
+        r"""Perform forward pass.
+
+        Args:
+            batch_sequences:
+                Batch of sequences which have been encoded by
+                `lmp.tokenizer.BaseTokenizer` with numeric type `torch.int64`.
+
+        Returns:
+            Logits for each token in sequences with numeric type `torch.float32`.
+        """
+        # 將 batch_sequences 中的所有 token_id 經過 embedding matrix
```

Comment:
Remove chinese comments and replace it with english comments.
Only do this in `_base_res_rnn_model.py`, do not replace comments in `_base_rnn_model.py`.